src/Lecture2.hs

```diff
@@ -70,7 +80,16 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+removeAt :: Int -> [a] -> (Maybe a, [a])
+removeAt _ [] = (Nothing, [])
+removeAt index list
+  | index < 0 = (Nothing, list)
+  | otherwise = (element, headPart ++ tailPart)
+    where
+      headPart = take index list
+      (element, tailPart) = case drop index list of
```

Comment:
That's a correct implementation 👍🏻 
Alternatively, you can use a standard function [splitAt](https://hackage.haskell.org/package/base-4.16.0.0/docs/Prelude.html#v:splitAt) instead of using `take` and `drop` to avoid traversing the list prefix twice 🔃

Reply:
That's a good point.